### PR TITLE
fix(catalog-backend): Support full text searches across multiple fields

### LINKED ISSUE #16920: 🐛 Bug Report: Paginated entities endpoint filters text searches on sortField
### 📜 Description

When using the `by-query` endpoint, `fullTextFilterTerm` is added to the high level query, filtering on the key defined by `sortField.field`. This value should instead have a sane default and support `fullTextFilterFields` being passed to define what fields to filter on.

### 👍 Expected behavior

I should be able to filter by text across multiple fields, not just the sort field.

### 👎 Actual Behavior with Screenshots

I can't sort across multiple fields and can only define one at a time.

### 👟 Reproduction steps

Run `GET http://localhost:7007/api/catalog/entities/by-query?filter=kind=component&limit=5&fullTextFilterTerm=play` in the dev environment. Nothing is returned, then try running this query `http://localhost:7007/api/catalog/entities/by-query?filter=kind=component&limit=5&fullTextFilterTerm=play&orderField=metadata.name,asc` and expected values are returned.

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by sennyeya**
## Hey, I just made a Pull Request!

Fixes #16920. `fullTextFilterFields` is unused and `fullTextFilterTerm` is currently filtering on the values defined by the `sortOrder.field` key, this PR seeks to add full text search functionality to the `by-query` endpoint for use with #16021.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by vinzscam — COMMENTED** [review]
We left this use case unimplemented on purpose, so thank you for taking care of this! 🚀

File: plugins/catalog-backend/src/service/DefaultEntitiesCatalog.test.ts [hunk 1]
```diff
@@ -1119,6 +1119,64 @@ describe('DefaultEntitiesCatalog', () => {
...
+    it.each(databases.eachSupportedId())(
+      'should filter the text results when sortOrder is not provided, %p',
+      async databaseId => {
+        await createDatabase(databaseId);
+
+        const names = ['lion', 'cat', 'atcatss', 'dog', 'dogcat', 'aa', 's'];
+        const entities: Entity[] = names.map(name =>
+          // Need a stable search since default filtering is by uid, and those get generated on the fly
+          //  during the test case.
+          entityFrom(name, { uid: name }),
```
[the remark this answers is not in the document]
Comment (sennyeya) [reply plugins/catalog-backend/src/service/DefaultEntitiesCatalog.test.ts]: Updated!

File: plugins/catalog-backend/src/service/DefaultEntitiesCatalog.test.ts [hunk 2]
```diff
@@ -1119,6 +1119,64 @@ describe('DefaultEntitiesCatalog', () => {
...
+        });
+
+        const filter = {
+          key: 'spec.should_include_this',
+        };
+
+        const request: QueryEntitiesInitialRequest = {
+          filter,
+          limit: 100,
+          fullTextFilter: { term: 'cAt ', fields: ['metadata.name'] },
```
[the remark this answers is not in the document]
Comment (sennyeya) [reply plugins/catalog-backend/src/service/DefaultEntitiesCatalog.test.ts]: Updated!

**Review by vinzscam — APPROVED** [review]
looks amazing! 🌟

**Review by freben — APPROVED** [review]
Let's go!

**Pull request merged**